### PR TITLE
Feature/all config

### PR DESCRIPTION
**Pull request by berkes**
This allows all jitisi-meet-config variables to be set from within this role.

I probably removes the need to install jitisi-meet-config deb package; and removes the need to run debconf on that, but since we move the file generated there, into a backup, we do create a nice reference. 

Please let me know if you wish me to remove debconf setup and the installation of that package.

### CODE REVIEW

**Comment by cimnine** [comment]
Thank you for your PR.

I like the idea, that the Ansible role can be used to configure Jitsi. What I don't like about your approach is that the maintainers of this role (currently mainly me) would have to keep the settings and the defaults in sync with every release of Jitsi Meet.

Hence I would prefer a solution which would still allow Jitsi to be configured in the Ansible role, but which would not require the synchronization work.

I was thinking that, instead of replacing them altogether, some kind of patching of the files that ship with the `jitisi-meet-config` would have to take place.

What's your opinion on this?

**Comment by berkes** [comment]
I am not familiar enough with ansible to know whether patching a json file is stable and practical. But I did see this is possible.

My ansible "philosophy" in my environments, is that there's always only one canonical source of a file: I dislike merging files from various sources. I avoid "lineininfile" as much as possible, have very bad experiences with "blockinfile" etc.
Personally, I want to know *exactly* who is providing the *entire* file and generally simply copy it over as template into my projects: to avoid conflicts, duplication, syntax errors and so on.

This is why I chose the current setup, where I to leave it to the implementor; the user of your role; to provide the proper variables: I simply render the entire struct provided by the user, as JSON.

I understand your point about having to keep the "config" in sync with jitisi, so, My personal preference aside:

Would you prefer to have me rewrite this with:

* lineinfile
* Use `jq` to patch the file: <https://stackoverflow.com/a/53546229/73673>
* Use ansible-jsonpatch: <https://github.com/ParticleDecay/ansible-jsonpatch>
* Or, use something else?

**Comment by cimnine** [comment]
I see your points. And I like your argument.

What about a compromise here? We'll add a flag, let's say `jtisi_manage_config: true|false`.
If it's `true`, then it's this roles' job to manage the config.

By default, `jtisi_manage_config` would be `false`. But if it's `true`, the following would execute:

```yaml
- when: jitsi_manage_config|bool    # check for jitsi_manage_config
  name: Configure Jitsi
  import_tasks: jitsi/config.yml
```

This way, we don't have to follow upstream, because by default, we don't mangle with what comes by default from upstream.

But it's still possible to configure the configuration using this Ansible role.

(Btw, I'm not particular happy with the variable name `jitsi_manage_config`. I've also thought about `jitsi_managed_config` or `jits_config_management`. Feel free to choose either or propose your own.)

**Comment by berkes** [comment]
Good idea and very clear.

I will implement this, but am a bit busy, so it might take a few days before it lands in this PR.

As for the name: For now i'll go with `use_custom_jitsi_config_vars` because that is not a double negative (pun intended), and it is a clear instruction to both the reader of the vars and to the roles: "use this instead of that".

**Comment by berkes** [comment]
@cimnine : I've implemented your suggestion and documented this behaviour in README.

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
```yaml
```

File: README.md
Comment (cimnine) [inline README.md]: I suggest that we put the default file here. I.e. what currently ships with `jitsi-web-config`. Or put differently: What you have added in the `defaults/main.yml` for `jitsi_config`, minus the commented lines.

So that the example becomes a _complete_ example, even though it's minimal. Because I believe incomplete examples are not helpful for beginners.

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
updating `jitsi-meet` through `apt`. The upstream project is not very communicative about this.
```

Could you keep each sentence together on one line and only split lines after a comma or point?

E.g.:

```markdown
NOTE:
The requirements of `config.js` may change at any moment when updating `jitsi-meet`.
The upstream project is not very communicative about this.
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
NOTE:
The requirements of `config.js` may change at any moment when
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
Check with any CHANGELOG and announcements of the _Jitsi Meet_ project about newly required values
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
NOTE:
The structure is case-sensitive and follows the exact variable names of the
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
settings in `config.js`. So, for example, `webrctIceUdpDisable` is very different to
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
(e.g. it mixes `URL` and `Url` at random), so pay attention to the exact name.
```

File: defaults/main.yml
Comment (cimnine) [inline defaults/main.yml]: I was hoping to start the variable name with `jitsi_`. All nginx related variables start with `nginx_`, the apt related ones with `apt_`, etc. But I see that `verb_something` is much more natural. How about shortening yours a bit:

```suggestion
managed_jitsi_config: false
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
_Jitsi Meet_ can be configured from this Ansible role. To do so, the
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
file `/etc/jitsi/meet/{{ jitsi_domain }}-config.js` from the upstream package will be replaced by a file managed by this
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion
To enable this behaviour define the variable `use_custom_jitsi_config_vars` and set it to `yes`.
```

File: README.md
Comment (cimnine) [inline README.md]:
```suggestion

I.e. add the last line of the following example to your `jitsi.ini`:

```ini
# jitsi.ini
[jitsi]
my-jitsi-server.com jitsi_domain=my-jitsi-server.com certbot_admin_email=admin@my-jitsi-server.com
[jitsi:vars]
ansible_user=ubuntu
ansible_become=yes
apt_mirror=http://archive.ubuntu.com/ubuntu # change to the mirror you already use
certbot_enabled=yes
jitsi_nat=no
use_custom_jitsi_config_vars=yes  # manage your jitsi config through this role

` ` ` (remove the spaces between the backticks)
```

**Review by cimnine — COMMENTED** [review]
Thanks for the updated PR. I really like the documentation part! I have left some comments. Feel free to leave a message wherever you disagree.

Could you also add the `use_custom_jitsi_config_vars` (or whatever it will be called) to the `## Role Variables` section in the beginning of the `README.md`? And a short text that tells the reader to find more information further below.

File: defaults/main.yml
Comment (berkes) [reply defaults/main.yml]: Myself I always ensure variables are namespaced instead. So not `nginx_foo` but rather `nginx.foo`. The first makes for clumsy and often very long variable names. `certbot_acme_link_use_certbot` and nonense that like that. 

I'll change it into either `managed_jitsi_config` or `jitsi_managed_config`, just let me know what you prefer.

File: README.md
Comment (berkes) [reply README.md]: I'll do that. 

however, the reasons that I did not, was threefold:

* I want to make sure people read the defaults.
* I want to make sure that copy-pasting this version probably breaks your setup. 
* I want to ensure we have only one place to maintain the required vars. If we copy it to README, that must be maintained as well as defaults.yml. Forget one, and you introduce copy-paste errors for users.

Comment (cimnine) [reply README.md]: > I want to ensure we have only one place to maintain the required vars. If we copy it to README, that must be maintained as well as defaults.yml. Forget one, and you introduce copy-paste errors for users.

That's a good point. And that's the tradeoff. I hope in this case that the bare minimum does not change too often.

File: defaults/main.yml
Comment (cimnine) [reply defaults/main.yml]: Ah, that would have been the way to go. I guess it's too late to change it now 🙈 `managed_jitsi_config` it is. Thanks for the hint! Much appreciated!

File: README.md
Comment (berkes) [reply README.md]: > does not change too often.

Here's for hoping: https://github.com/jitsi/jitsi-meet/commits/master/config.js

**Review by cimnine — APPROVED** [review]
Looks good. I'll merge when I'm not as tired as I am now.

**Pull request merged**
